From 3b87a74c32bf0724d1037bb079ceb1be43e5c868 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:24:56 -0400
Subject: [s390] pci: Fix single MSI only check

Message-id: <20140719042455.19438.76354.stgit@gir.stowe>
Patchwork-id: 85983
O-Subject: [RHEL7.1 PATCH 21/81] s390/PCI: Fix single MSI only check
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: a384c8924a8be3f5cf7dac06c04e9dd4a78b41df

commit a384c8924a8be3f5cf7dac06c04e9dd4a78b41df
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Mon Dec 16 09:34:54 2013 +0100

    s390/PCI: Fix single MSI only check

    Multiple MSIs have never been supported on s390 architecture, but the
    platform code fails to report single MSI only.

    Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/s390/pci/pci.c b/arch/s390/pci/pci.c
index 0cece14..189b49c 100644
--- a/arch/s390/pci/pci.c
+++ b/arch/s390/pci/pci.c
@@ -438,6 +438,8 @@ int arch_setup_msi_irqs(struct pci_dev *pdev, int nvec, int type)
 
  if (type != PCI_CAP_ID_MSIX && type != PCI_CAP_ID_MSI)
   return -EINVAL;
+ if (type == PCI_CAP_ID_MSI && nvec > 1)
+  return 1;
  msi_vecs = min(nvec, ZPCI_MSI_VEC_MAX);
 
  /* Allocate adapter summary indicator bit */
-- 
1.7.1